pymysql/connections.py

```diff
@@ -1191,4 +1227,70 @@ def _get_descriptions(self):
         assert eof_packet.is_eof_packet(), 'Protocol error, expecting EOF'
         self.description = tuple(description)
 
+
+class LoadLocalFile(object):
+    def __init__(self, filename, connection):
+        self.filename = filename
+        self.connection = connection
+
+    def _print_warnings(self):
+        self.connection._execute_command(COMMAND.COM_QUERY, 'SHOW WARNINGS')
+        warnings = MySQLResult(self.connection)
+        warnings.read()
+        if warnings.rows:
+            warning_source = traceback.extract_stack()[0]
+            print("{0}:{1}: {2}".format(str(warning_source[0]), str(warning_source[1]), str(warning_source[3])))
+            for warning in warnings.rows:
+                print("  Warning: {0} in file '{1}'".format(warning[2], self.filename.decode('utf-8')))
+
+    def send_data(self):
+        """Send data packets from the local file to the server"""
+        if not self.connection.socket:
+            raise InterfaceError("(0, '')")
+
+        # sequence id is 2 as we already sent a query packet
+        seq_id = 2
+        try:
+            with open(self.filename, 'r') as open_file:
+                chunk_size = MAX_PACKET_LEN
+                prelude = b""
+                packet = b""
+                packet_size = 0
+
+                for line in open_file:
+                    line_length = len(line)
+                    format_str = '!{0}s'.format(line_length)
+                    format_str = format_str.encode(self.connection.encoding)
```

Comment:
Ref http://dev.mysql.com/doc/refman/5.1/en/load-data.html
MySQL uses character_set_database variable or "CHARACTER SET" clause.  Connection encoding should not affect file content.

Open file with 'rb' mode and use loop like:
```python
while True:
    chunk = open_file.read(chunk_size)
    if not chunk:
        break
    packet = struct.pack('<i', len(chunk))[:3] + int2byte(seq_id)
    self.connection._write_bytes(packet)
    seq_id += 1
```